service/search.js

```diff
@@ -21,13 +23,28 @@ function service( esclient, cmd, cb ){
       return cb( err );
     }
 
+    // in the case of query timeout the response body contains the
+    // property 'timed_out=true'.
+    // these responses contain partially processed results and so should
+    // be discarded.
+    // https://github.com/pelias/api/issues/1384
+    if( _.get(data, 'timed_out', false) === true ){
+      const err = new es.errors.RequestTimeout('request timed_out=true');
+      logger.error( `elasticsearch error ${err}` );
+      return cb( err );
+    }
+
     // map returned documents
     var docs = [];
     var meta = {
       scores: []
     };
 
-    if( data && data.hits && data.hits.total && Array.isArray(data.hits.hits)){
+    if (
+        _.has(data, 'hits') && 
```

Comment:
I think you can remove this line, since the line below it will end up running the comparison `0 > 0` if data.hits is undefined

Reply:
Yeah I was thinking that too

Reply:
ok so I ended up rewriting it with future though towards ES7 where they decided to make `hits.total` an object 🤷‍♂ 
anyway, it's nicer to not rely on checking another property which we don't even reference in the code.

I'll add it as a new commit for clarity.

Reply:
done, this what they changed it to BTW:

```js
"hits": {
        "max_score": 1.0,
        "total" : {
            "value": 2048,    
            "relation": "eq"  
        },
        "hits": ...
    }
```